### PR TITLE
Fix #5797: Ride previews going out of bounds.

### LINKED ISSUE #5797: Crash when clicking on hedge maze in ride construction window
**OS:** Win 7 64
**Version:** 0.0.7
**Commit/Build:**  ~~Whatever was the version that updated at around 2 PM CST (07/04/17)~~ 4e4022d

Clicked on the Hedge Maze (select a ride for construction window), game crashed and said to post the dump file.

**Dump file**

[dump.zip](https://github.com/OpenRCT2/OpenRCT2/files/1123406/dump.zip)

### PR DESCRIPTION
**Pull request by ZehM4tt**
This should prevent #5797 from crashing. To my understanding this crash happens if some scenery is not loaded which is referenced however.

### CODE REVIEW

File: src/openrct2/object_list.c [hunk 1]
```diff
@@ -134,8 +134,10 @@ void object_create_identifier_name(char* string_buffer, size_t size, const rct_o
  * ecx = entry_type
  */
 sint32 find_object_in_entry_group(const rct_object_entry* entry, uint8* entry_type, uint8* entry_index){
+    if ((entry->flags & 0xF) >= _countof(object_entry_groups)) {
```
Comment (janisozaur) [inline src/openrct2/object_list.c]: `countof`

Comment (Gymnasiast) [inline src/openrct2/object_list.c]: macOS also chokes on this.

**Comment by Gymnasiast** [comment]
Tested it. It does indeed prevent a crash when hovering over the track list.

**Comment by IntelOrca** [comment]
I think we should merge this as its good to code defensively. But yes we do need to make sure we import track designs correctly.

**Comment by Gymnasiast** [comment]
@IntelOrca It somewhat hides the problem. But having the game crash outright on the track list is more than just inconvenient, so I'll merge it then.

**Comment by IntelOrca** [comment]
@ZehM4tt make sure your `Fix #5797: ...` is in your commit message.

**Pull request merged**